R/transformAbundance.R

```diff
@@ -359,6 +366,13 @@ setMethod("ZTransform", signature = c(x = "SummarizedExperiment"),
 
 .get_ztransformed_table <- function(assay, pseudocount, scalingFactor){
 
+    # If assay contains missing values
+    if(any(is.na(assay))){
+        # Missing values are replaced with value 0
+        assay[which(is.na(assay))] <- 0
+        warning("Table contains missing values. Missing values are replaced with value 0.")
+    }
+
```

Comment:
I think this might not the way to do it, since the result is actually changed by adding zeros. The solution probably hast to be aware of NAs and filter it out at the calculation step.

The idea would be to have a factor for subsetting the data in steps, which cannot handle NAs, like this:
```
f <- is.na(data)
data[f] <- FUN(data[f])
```

Reply:
I agree, this is potentially risky and should be changed now with the update.

Reply:
(or it could be an option to replace NAs by zeroes or imputation but perhaps it is more clear if the user can choose to do that outside of the function)